CleverTapSDKTests/InApps/CustomTemplates/SystemAppFunction/CTSystemTemplateActionHandlerTest.m

```diff
@@ -0,0 +1,81 @@
+//
+//  CTSystemTemplateActionHandlerTest.m
+//  CleverTapSDKTests
+//
+//  Created by Nishant Kumar on 15/04/25.
+//  Copyright © 2025 CleverTap. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+#import <OCMock/OCMock.h>
+#import "CTSystemTemplateActionHandler.h"
+#import "CTPushPrimerManagerMock.h"
+#import "InAppHelper.h"
+#import "CTUIUtils.h"
+
+API_AVAILABLE(ios(10.0))
+@interface CTSystemTemplateActionHandlerTest : XCTestCase
+
+@property (nonatomic, strong) CTSystemTemplateActionHandler *actionHandler;
+@property (nonatomic, strong) CTPushPrimerManagerMock *pushPrimerManager;
+@property (nonatomic, strong) InAppHelper *helper;
+
+@end
+
+@implementation CTSystemTemplateActionHandlerTest
+
+- (void)setUp {
+    self.actionHandler = [[CTSystemTemplateActionHandler alloc] init];
+    
+    self.helper = [InAppHelper new];
+    CTDispatchQueueManager *queueManager = [[CTDispatchQueueManager alloc] initWithConfig:self.helper.config];
+    self.pushPrimerManager = [[CTPushPrimerManagerMock alloc] initWithConfig:self.helper.config
+                                                         inAppDisplayManager:self.helper.inAppDisplayManager
+                                                        dispatchQueueManager:queueManager];
+    [self.actionHandler setPushPrimerManager:self.pushPrimerManager];
+    
+    // Mock CTUIUtils openURL method to not open any url.
+    id mockCTUIUtils = OCMClassMock([CTUIUtils class]);
```

Comment:
Set the mock to a property so we can stop mocking once the test is done.
```
@property (nonatomic) id mockCTUIUtils;
```

Reply:
Done